image-hd.c

```diff
@@ -551,7 +578,20 @@ static int hdimage_setup(struct image *image, cfg_t *cfg)
 					"when using a 'autoresize' partition\n");
 			return -EINVAL;
 		}
-		if (hd->gpt) {
+		if (part->partition_type_uuid && !(hd->table_type & TYPE_GPT)) {
+			image_error(image, "part %s: 'partition-type-uuid' is only valid for gpt and hybrid partition-table-type\n",
+					part->name);
+			return -EINVAL;
+		}
+		if (part->partition_type && !(hd->table_type & TYPE_MBR)) {
+			image_error(image, "part %s: 'partition-type-uuid' is only valid for mbr and hybrid partition-table-type\n",
```

Comment:
I think this error text should be `partition-type`, not `partition-type-uuid`.

Reply:
Indeed. Thanks for the review. Fixed.